### PR TITLE
Hateoas(User&Product)Hateoas installed

### PR DESCRIPTION
**Pull request by VblDigital**
Hateoas-bundle installed.
First link create for route "view_users" -> doesn't work

### CODE REVIEW

File: src/Entity/User.php [hunk 1]
```diff
@@ -3,11 +3,21 @@
...
+use Hateoas\Configuration\Annotation as Hateoas;
 
 /**
  * @ORM\Entity(repositoryClass="App\Repository\UserRepository")
+ * @ORM\Table()
+ * @Hateoas\Relation(
+ *      "self",
+ *      href = @Hateoas\Route(
+ *          "view_users",
+ *          parameters = { "id" = "expr(object.getId())" },
```
Comment (blanselle) [inline src/Entity/User.php]: Tu appelles la route view_users mais tu l'as nommé plus haut view_user

**Review by blanselle — APPROVED** [review]

File: src/Controller/UserController.php [hunk 2]
```diff
@@ -104,13 +91,14 @@ public function newUser(User $user, $clientId, EntityManagerInterface $manager,
      * @Rest\Put(
      *     path = "/users/{userId}",
      *     name = "modify_user")
+     * @View(serializerGroups={"credentials"})
```
Comment (blanselle) [inline src/Controller/UserController.php:94]: @View(serializerGroups={"credentials"}, statusCode=201)

File: src/Controller/UserController.php [hunk 3]
```diff
@@ -120,38 +108,40 @@ public function modifyUser(User $user, $userId, UserRepository $repository, Enti
...
+            ->setPassword(password_hash($user->getPassword(), PASSWORD_BCRYPT))
             ->setEmail($user->getEmail());
-        $hash = password_hash($user->getPassword(), PASSWORD_BCRYPT);
-        $registeredUser->setPassword($hash);
 
         $manager->persist($registeredUser);
         $manager->flush();
 
-        return $this->view($registeredUser, Response::HTTP_CREATED);
+        return $this->view($registeredUser, Response::HTTP_ACCEPTED);
```
Comment (blanselle) [inline src/Controller/UserController.php:120]: return $registeredUser

File: src/Controller/UserController.php [hunk 4]
```diff
@@ -104,13 +91,14 @@ public function newUser(User $user, $clientId, EntityManagerInterface $manager,
      * @Rest\Put(
      *     path = "/users/{userId}",
```
Comment (blanselle) [inline src/Controller/UserController.php:92]: /users/{id}

File: src/Controller/UserController.php [hunk 5]
```diff
@@ -120,38 +108,40 @@ public function modifyUser(User $user, $userId, UserRepository $repository, Enti
...
 
-        return $this->view($registeredUser, Response::HTTP_CREATED);
+        return $this->view($registeredUser, Response::HTTP_ACCEPTED);
     }
 
     /**
      * @Rest\Delete(
      *     path = "/users/{userId}",
      *     name = "delete_user")
+     * @View(serializerGroups={"credentials"})
```
Comment (blanselle) [inline src/Controller/UserController.php:127]: @View(serializerGroups={"credentials"}, statusCode=204)

File: src/Controller/UserController.php (hunk 2 above)
Comment (VblDigital) [reply src/Controller/UserController.php:94]: Done

File: src/Controller/UserController.php (hunk 3 above)
Comment (VblDigital) [reply src/Controller/UserController.php:120]: Done

File: src/Controller/UserController.php (hunk 5 above)
Comment (VblDigital) [reply src/Controller/UserController.php:127]: Done

**Pull request merged**